src/main/java/org/zalando/nakadi/validation/SchemaEvolutionService.java

```diff
@@ -77,19 +86,42 @@ public EventType evolve(final EventType original, final EventTypeBase eventType)
 
         if (incompatibility.isPresent()) {
             throw new InvalidEventTypeException(incompatibility.get().getReason());
-        } else {
-            final List<SchemaChange> changes = schemaDiff.collectChanges(schema(original), schema(eventType));
+        }
+
+        final List<SchemaChange> changes = schemaDiff.collectChanges(schema(original), schema(eventType));
 
-            final Version.Level changeLevel = semanticOfChange(changes);
+        final Version.Level changeLevel = semanticOfChange(changes);
 
-            if (eventType.getCompatibilityMode() == CompatibilityMode.COMPATIBLE && changeLevel == MAJOR) {
+        validateCompatibleModeChanges(original, changes, changeLevel);
+        validateCompatibilityModeMigration(original, eventType, changes);
+
+        return this.bumpVersion(original, eventType, changeLevel);
+    }
+
+    private void validateCompatibilityModeMigration(final EventType original, final EventTypeBase eventType,
+                                                    final List<SchemaChange> changes) throws InvalidEventTypeException {
+        if (original.getCompatibilityMode() == CompatibilityMode.FIXED
+            && eventType.getCompatibilityMode() == CompatibilityMode.COMPATIBLE) {
+            final List<SchemaChange.Type> changesTypes = changes.stream().map(SchemaChange::getType)
+                    .collect(Collectors.toList());
+            if (!FIXED_TO_COMPATIBLE_ALLOWED_CHANGES.containsAll(changesTypes)) {
                 final String errorMessage = changes.stream()
-                        .filter(change -> MAJOR.equals(changeToLevel.get(change.getType())))
+                        .filter(change -> !FIXED_TO_COMPATIBLE_ALLOWED_CHANGES.contains(change.getType()))
                         .map(this::formatErrorMessage)
                         .collect(Collectors.joining(", "));
                 throw new InvalidEventTypeException("Invalid schema: " + errorMessage);
             }
-            return this.bumpVersion(original, eventType, changeLevel);
+        }
+    }
+
+    private void validateCompatibleModeChanges(final EventType original, final List<SchemaChange> changes,
+                                               final Version.Level changeLevel) throws InvalidEventTypeException {
+        if (original.getCompatibilityMode() == CompatibilityMode.COMPATIBLE && changeLevel == MAJOR) {
+            final String errorMessage = changes.stream()
+                    .filter(change -> MAJOR.equals(changeToLevel.get(change.getType())))
```

Comment:
Why do you only show major incompatibilities? AFAIU there could be other levels.